### PR TITLE
reposurgeon 3.21

### PR DESCRIPTION
**Pull request by dunn**

### CODE REVIEW

**Comment by dunn** [comment]
Failures across the board, but the test worked locally—not sure what the difference is.

File: Library/Formula/reposurgeon.rb [hunk 1]
```diff
@@ -1,18 +1,39 @@
...
+
+  test do
+    (testpath/".gitconfig").write <<-EOS.undent
+      [user]
+        name = Real Person
+        email = notacat@hotmail.cat
+      EOS
+    system "git", "init"
+
+    require "open3"
```
Comment (xu-cheng) [inline Library/Formula/reposurgeon.rb]: Let's try to avoid using `open3`. Instead you can use `pipe_output`

Comment (dunn) [inline Library/Formula/reposurgeon.rb]: I don't think it allows commands to be piped in.  When I try `assert_match "found git repository", pipe_output("#{bin}/reposurgeon", "read")` I get this:
```
🐙  brew test -v reposurgeon
Testing reposurgeon
==> git init
Initialized empty Git repository in /private/tmp/reposurgeon20150510-795-1xv6t0v/.git/
==> /usr/local/Cellar/reposurgeon/3.21/bin/reposurgeon
reposurgeon: fatal exception in popen_or_die.
reposurgeon:
```

Comment (xu-cheng) [inline Library/Formula/reposurgeon.rb]: That's weird. `pipe_output` uses `IO.popen` which it's not that different from `open3`.

Can you try this `pipe_output("script -q /dev/null #{bin}/reposurgeon", "...")`?

Comment (dunn) [inline Library/Formula/reposurgeon.rb]: That fails to complete, as does this:
```ruby
    IO.popen("#{bin}/reposurgeon") do |pipe|
      pipe.puts "read\n"
      pipe.close_write
      assert_match "found git repository", pipe.get
    end
```

Comment (xu-cheng) [inline Library/Formula/reposurgeon.rb]: fair enough.

Comment (dunn) [inline Library/Formula/reposurgeon.rb]: (`pipe.get` should have been `pipe.gets`, but that doesn't change the result here.)

**Comment by dunn** [comment]
looks like this is the same situation as https://github.com/Homebrew/homebrew/pull/34165#discussion_r22342214

**Comment by dunn** [comment]
(for reference, the new test failure: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/25967/version=mountain_lion/testReport/junit/brew-test-bot/mountain_lion/test_reposurgeon/)

**Comment by xu-cheng** [comment]
Thanks @dunn

**Pull request closed without merging**